src/App.js

```diff
@@ -1,26 +1,28 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
 
-export class App extends Component {
-  state = {
-    movies: data,
-  };
+export const App = () => {
+  const [movies, setMovies] = useState(data);
 
-  render() {
-    const { movies } = this.state;
+  function addMovie(movie) {
+    setMovies(currentMovies => [...currentMovies, movie]);
```

Comment:
```suggestion
    setMovies([...movies, movie]);
```
не необходимости использовать callback если prevState находиться в области видимости